controllers/utils.go

```diff
@@ -1,5 +1,23 @@
 package controllers
 
+import (
+	"fmt"
+
+	gogithub "github.com/google/go-github/v37/github"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RunnerCache holds cached returns from github runner API calls
+// - default time for cache is 60 * time.Seconds()
+type RunnerCache struct {
```

Comment:
The RunnerCache structure is missing the dimensions where you have multiple runners, per repo, orgs or enterprise. In the current implementation, it looks like it would only work with one type of runners in a single controller.

Instead, you could have a `RunnerCacheMap` structure that would point to a `RunnerCache` for each runner group.

Reply:
Gotcha!  I will take a look at that.